node/src/store/indexed_db_store.rs

```diff
@@ -280,6 +370,38 @@ async fn get_head_from_database(db: &Rexie) -> Result<ExtendedHeader> {
         .map_err(|e| StoreError::CelestiaTypes(e.into()))
 }
 
+/// Return the height of the lowest unsampled header from the database, starting
+/// from `last_known_unsampled` offset
+async fn get_next_unsampled_height_from_database(
+    db: &Rexie,
+    last_known_unsampled: u64,
+) -> Result<u64> {
+    let tx = db.transaction(&[SAMPLING_STORE_NAME], TransactionMode::ReadOnly)?;
+    let store = tx.store(SAMPLING_STORE_NAME)?;
+
+    let pending_headers_keyrange = KeyRange::lower_bound(&to_value(&last_known_unsampled)?, false)?;
+    let pending_sampled_headers = store
+        .get_all(
+            Some(&pending_headers_keyrange),
+            None,
+            None,
+            Some(Direction::Next),
+        )
+        .await?;
```

Comment:
this is a bit dangerous. If I understand correctly, this will load all headers from the indexeddb after some index. Can this result in loading all the headers after we release the DASer? Since someone can be in sync with mainnet but have nothing dased yet.

I think we should have another loop there, and paginate through headers. Let's add some limit on this like 1000 headers at the time in each iteration

Reply:
or maybe even by 10 (50, 100?) headers at the time? Since from what I see this is gonna be called each time a header is marked sampled, and it's likely we only need to check very few next headers in a regular case